File: app/src/main/java/com/arash/altafi/lazyload/java/api/ApiService.java
package com.arash.altafi.lazyload.java.api;

import com.arash.altafi.lazyload.java.model.ResponseCelebritiesJava;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface ApiService {

    @GET("lazy_load/LazyLoad.php")
    Single<List<ResponseCelebritiesJava>> lazyLoad(@Query("name") String name, @Query("page") int pageCount, @Query("per_page") int perPage);

}

File: app/src/main/java/com/arash/altafi/lazyload/kotlin/api/ApiService.kt
package com.arash.altafi.lazyload.kotlin.api

import com.arash.altafi.lazyload.kotlin.model.ResponseCelebrities
import io.reactivex.Single
import retrofit2.http.*

interface ApiService {

    @GET("lazy_load/LazyLoad.php")
    fun lazyLoad(
        @Query("name") name: String,
        @Query("page") pageCount: Int,
        @Query("per_page") perPage: Int
    ): Single<List<ResponseCelebrities>>

    @FormUrlEncoded
    @POST("lazy_load/LazyLoad.php")
    fun lazyLoad2(
        @Field("name") name: String,
        @Query("page") pageCount: Int,
        @Query("per_page") perPage: Int
    ): Single<List<ResponseCelebrities>>

}